src/profile_service.rs

```diff
@@ -104,7 +110,8 @@ fn test_custom_profile() {
     let profile_path = String::from(profile_dir.into_path()
                                     .to_str().unwrap());
 
-    let profile = ProfileService::new(Some(profile_path.to_owned()));
+    let profile = ProfileService::new(ProfilePath::Custom(profile_path
+                                                          .clone()));
```

Comment:
Oh I see what you mean. That would work too. Although the convention seems to be not assigning state variables for this ownership management stuff. A style guide would help here, but I guess you guys are still figuring that out.